File: level1/746/minCostClimbStairs.c
#include <stdio.h>

int min(int a, int b)
{   
    return(a < b ? a : b);
}

int minCostClimbingStairs(int* cost, int costSize)
{   
    int sum = 0;
    for (int i = 2; i < costSize; i++)
    {
        cost[i] += min(cost[i - 1], cost[i - 2]);
    }
    sum = min(cost[costSize-1], cost[costSize-2]);
    return (sum);
}


int main(void)
{
    int n[] = {1,100,1,1,1,100,1,1,100,1};
    int size = sizeof(n) / sizeof(int);
    int mincost = minCostClimbingStairs(n, size);
    printf("Result: %d\n", mincost);
    return (0);
}

File: level1/746/minCostClimbStairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        n = len(cost)
        if n == 1:
            return cost[0]
        dp = [None] * n
        dp[0] = cost[0] 
        dp[1] = cost[1]
        for i in range(2, n):
            dp[i] = min(dp[i - 1], dp[i - 2]) + cost[i]
        return min(dp[n - 1], dp[n -2])